src/hooks/use-log-event.js

```diff
@@ -0,0 +1,30 @@
+import debounce from 'lodash.debounce';
+import React from 'react';
+import { useEventProperties } from './use-event-properties';
+import { useInstance } from './use-instance';
+
+export const useLogEvent = (instanceName, eventProperties, debounceInterval) => {
+  const instance = useInstance(instanceName);
+  const inheritedEventProperties = useEventProperties(eventProperties);
+
+  return React.useMemo(() => {
+    const logEvent = (eventType, eventProperties, callback) => {
+      if (instance) {
+        instance.logEvent(
+          eventType,
+          {
+            ...inheritedEventProperties,
+            ...(eventProperties || {}),
```

Comment:
`|| {}` is unnecessary for *object* spread operator, undefined & null are just swallowed